src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -122,7 +181,46 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
 
         protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventTypes, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            if (eventTypes.HasFlag(PinEventTypes.Rising) || eventTypes.HasFlag(PinEventTypes.Falling))
+            {
+                LibGpiodDriverEventHandler eventHandler = _pinNumberToEventHandler.GetOrAdd(pinNumber, PopulateEventHandler);
+
+                if (eventTypes.HasFlag(PinEventTypes.Rising))
+                {
+                    eventHandler.ValueRising += callback;
+                }
+                else if (eventTypes.HasFlag(PinEventTypes.Falling))
+                {
+                    eventHandler.ValueFalling += callback;
+                }
+
+                bool eventOccured = false;
+                void callback(object o, PinValueChangedEventArgs e)
+                {
+                    eventOccured = true;
+                }
+
+                WaitForEventResult(cancellationToken, eventHandler.CancellationTokenSource.Token, ref eventOccured);
+                RemoveCallbackForPinValueChangedEvent(pinNumber, callback);
+
+                return new WaitForEventResult
+                {
+                    TimedOut = !eventOccured,
+                    EventTypes = eventTypes
+                };
+            }
+            else
+            {
+                throw ExceptionHelper.GetArgumentException(ExceptionResource.InvalidEventType);
+            }
+        }
+
+        private void WaitForEventResult(CancellationToken sourceToken, CancellationToken parentToken, ref bool eventOccured)
+        {
+            while (!(sourceToken.IsCancellationRequested || parentToken.IsCancellationRequested || eventOccured))
+            {
+                Thread.Sleep(1);
```

Comment:
this feels like it's going to burn the CPU with such a low value.

/cc @stephentoub

Reply:
Do we need to spin?  There's nothing we can block waiting for, a callback we can receive, etc.?

Reply:
> this feels like it's going to burn the CPU with such a low value.

Put a value used in lbgpiod samples (same as we use for hardware level timeout https://github.com/dotnet/iot/pull/236/files#diff-dc75afeeda145b47dfc2bd34d7f04d21R171)

> Do we need to spin? There's nothing we can block waiting for, a callback we can receive, etc.?

Reused logic used for AddCallbackForPinValueChangedEvent(...). It can be implemented separately and listen for event directly from hardare. But in that case the pin could only be used for that purpose. In other words if the pin have been used for listening other events this will just add another callback.